port/common/posix/oc_poll.c

```diff
@@ -0,0 +1,64 @@
+/****************************************************************************
+ *
+ * Copyright (c) 2023 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef _WIN32
+
+// make ppoll() available
+#define _GNU_SOURCE
+
+#include "port/common/posix/oc_poll_internal.h"
+#include "oc_clock_util.h"
+
+#include <unistd.h>
+
+static int
```

Comment:
`int` is wired when you processing events because `on_event` can stop processing so you don't process all n so maybe change it to enum like: `error,timeout,event` ?